Dockerfile

```diff
@@ -4,6 +4,11 @@
 # change is that the Hugo version is now an overridable argument rather than a fixed
 # environment variable.
 
+# Bump this by 1 whenever you change the Dockerfile below, e.g. if
+# `DOCKERFFILE_VERSION=1` then change this to `DOCKERFILE_VERSION=2` when you
+# change something else in this file.
+# DOCKERFILE_VERSION=0
```

Comment:
There are other ways we could accomplish this part, but basically we just need some sub-revision beyond just the hugo version, since the image/build also changes.

This way means that whenever someone wants to change how we go from HUGO_VERSION => image, they should bump this version.